### PR TITLE
Add intents for searching MangaPlus, use apiUrl over baseUrl

### PR DESCRIPTION
**Pull request by ZaneHannanAU**
Attempt at adding intent support to mangaplus extension.

### CODE REVIEW

**Comment by ZaneHannanAU** [comment]
Please note that this is my first time writing Kotlin (or well... any language outside of rust/c/js/ts). I replaced the data with the best translation I could set up (using JSON module); perhaps to use later on.

File: src/en/mangaplus/src/eu/kanade/tachiyomi/extension/en/mangaplus/MangaPlus.kt [hunk 1]
```diff
@@ -254,11 +272,13 @@ class MangaPlus : HttpSource() {
...
         private val JSON_PARSER by lazy { JsonParser() }
 
+        private const val PREFIX_ID_SEARCH = "id:"
+
         private const val IMAGE_DECRYPT_SRC = """
             function hex2bin(hex) {
                 return new Uint8Array(hex.match(/.{1,2}/g)
                         .map(function (x) { return parseInt(x, 16) }));
-            }
+            };
```
Comment (arkon) [inline src/en/mangaplus/src/eu/kanade/tachiyomi/extension/en/mangaplus/MangaPlus.kt]: Remove these extra semicolons, they're not needed for functions.

**Comment by ZaneHannanAU** [comment]
... how can I convert an `Observable<Observable<MangasPage>>` into an `Observable<MangasPage>`?

**Comment by ZaneHannanAU** [comment]
Cool. Ready for uh thing

**Pull request closed without merging**